internal/server/data/migrations.go

```diff
@@ -1173,3 +1174,70 @@ func addUserPublicKeyUserIDIndex() *migrator.Migration {
 		},
 	}
 }
+
+func addGrantsSubjectID() *migrator.Migration {
+	return &migrator.Migration{
+		ID: "2023-01-12T17:00",
+		Migrate: func(tx migrator.DB) error {
+			if !migrator.HasColumn(tx, "grants", "subject") {
+				return nil
+			}
+
+			// Step 1 - Add the new columns
+			_, err := tx.Exec(`
+				ALTER TABLE grants ADD COLUMN IF NOT EXISTS subject_id bigint;
+				ALTER TABLE grants ADD COLUMN IF NOT EXISTS subject_kind smallint;
+			`)
+			if err != nil {
+				return err
+			}
+
+			// Step 2- Migrate the data
+			type grantSubject struct {
+				ID              uid.ID
+				OriginalSubject string
+			}
+
+			rows, err := tx.Query(`SELECT id, subject FROM grants`)
+			if err != nil {
+				return err
+			}
+			grants, err := scanRows(rows, func(g *grantSubject) []any {
+				return []any{&g.ID, &g.OriginalSubject}
+			})
+			if err != nil {
+				return err
+			}
+
+			for _, grant := range grants {
+				var kind int
+				var subjectID uid.ID
+				if strings.HasPrefix(grant.OriginalSubject, "i:") {
+					kind = 1
+				} else {
+					kind = 2
+				}
+				subjectID, err = uid.Parse([]byte(grant.OriginalSubject[2:]))
+				if err != nil {
+					return err
+				}
+
+				_, err = tx.Exec(`UPDATE grants SET subject_id=?, subject_kind=? WHERE id = ?`,
```

Comment:
Iterating over every grant and doing an individual update is relatively slow. We can see how long this takes in dev. I suspect it'll be fine at our current scale.